src/inference/dev_api/openvino/runtime/iremote_tensor.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief OpenVINO Runtime IRemoteTensor interface
+ * @file openvino/runtime/iremote_tensor.hpp
+ */
+
+#pragma once
+
+#include "openvino/core/except.hpp"
+#include "openvino/runtime/common.hpp"
+#include "openvino/runtime/itensor.hpp"
+#include "openvino/runtime/tensor.hpp"
+
+namespace ov {
+
+class OPENVINO_RUNTIME_API IRemoteTensor : public ITensor {
+public:
+    void* data(const element::Type& type = {}) const final {
+        OPENVINO_NOT_IMPLEMENTED;
```

Comment:
why not `deleted` as it's in public API?

Reply:
this method is virtual I cannot use `delete`. In the next PR I will try to replace it by `static_assert`